### PR TITLE
Tangled tree visualization

### PR DESCRIPTION
**Pull request by abhinavchobey**
# Related Issues / Pull Requests

List all related issues and/or pull requests if there are any.

# Description

Include a brief summary of the proposed changes.

# What changes are proposed in this pull request?

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected; for instance, 
      examples in this repository need to be updated too).
- [ ] This change requires a documentation update.

# Checklist:

- [ ] My code follows the style guidelines of this project (PEP-8 with Google-style docstrings).
- [ ] My code modifies existing public API, or introduces new public API, and I updated or wrote docstrings that
      uses Google-style formatting and any other formatting that is supported by mkdocs and plugins this project
      uses.
- [ ] I have commented my code.
- [ ] My code requires documentation updates, and I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: cmflib/cmfquery.py [hunk 1]
```diff
@@ -617,6 +617,46 @@ def get_one_hop_parent_executions(self, execution_id: t.List[int], pipeline_id:
...
+    def get_executions_with_execution_ids(self, exe_ids: t.List[int]):
+        """For list of execution ids it returns df with "id,Execution_type_name, Execution_uuid"
+        Args:
+            execution ids: List of execution ids.
+        Return:
+            ["id","Execution_type_name","Execution_uuid"]
+        """
+        df = pd.DataFrame()
+        executions = self.store.get_executions_by_id(exe_ids)
+        count = 0
```
Comment (annmary-roy) [inline cmflib/cmfquery.py:649]: You can use enumerate instead of using the counter

File: cmflib/cmfquery.py [hunk 2]
```diff
@@ -617,6 +617,46 @@ def get_one_hop_parent_executions(self, execution_id: t.List[int], pipeline_id:
...
+        for exe in executions:
+            temp_dict = {}
+            temp_dict['id'] = exe_ids[count]
+            d1 = self._transform_to_dataframe(exe, temp_dict)
+            df = pd.concat([df, d1], sort=True, ignore_index=True)
+            count +=1
```
Comment (annmary-roy) [inline cmflib/cmfquery.py:655]: Use enumerate rather than counter

File: cmflib/cmfquery.py (hunk 1 above)
Comment (abhinavchobey) [reply cmflib/cmfquery.py]: Updated with enumerate

File: cmflib/cmfquery.py (hunk 2 above)
Comment (abhinavchobey) [reply cmflib/cmfquery.py]: Updated with enumerate

File: server/app/query_tangled_lineage.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    query = cmfquery.CmfQuery(mlmd_path)
+    pipeline_id = query.get_pipeline_id(pipeline_name)
+    df=dict_of_exe_id[pipeline_name]
+    
+    #finding execution_id by comparing Execution_uuid (d09fdb26-0e9d-11ef-944f-4bf54f5aca7f) and uuid ('Prepare_u3tr')  
+    result = df[df['Execution_uuid'].str[:4] == uuid]   #result = df[id: "1","Execution_type_name", "Execution_uuid"]
+    execution_id=result["id"].tolist() 
+    parents_set = set()
+    queue = deque()  
+    pd.set_option("display.max_columns", None)
```
Comment (annmary-roy) [inline server/app/query_tangled_lineage.py:16]: Is this needed ?

**Review by annmary-roy — APPROVED** [review]

**Pull request merged**